Controllers/CHumidityController.cpp

```diff
@@ -0,0 +1,261 @@
+                                     IComChannel *p_comchannel)
+{
+    bool b_command_recognised = false;
+    ICommand::command_error_code_t result = ICommand::COMMAND_OK;
+    /**
+     * Query about current state of all active channels.
+     *
+     */
+    if (p_command->getName()->compare("?humidity") == 0)
+    {
+        sendStatus(p_comchannel);
+        b_command_recognised = true;
+    }
+
+    if (p_command->getName()->compare("humidity") == 0)
+    {
+        result = setHumidity(p_command);
+        b_command_recognised = true;
+    }
+
+    if (p_command->getName()->compare("heater") == 0)
+    {
+        result = overrideHeater(p_command);
+        b_command_recognised = true;
+    }
+    if (b_command_recognised)
+    {
+        switch (result)
+        {
+            case ICommand::COMMAND_OK:
+                //                p_comchannel->send("OK.\n");
+                break;
+            case ICommand::ERROR_ARG_COUNT:
+                p_comchannel->send("Wrong number of arguments.\n");
+                break;
+            case ICommand::ERROR_OUT_OF_BOUNDS:
+                p_comchannel->send("Argument out of bounds.\n");
+                break;
+            case ICommand::ERROR_TYPE_MISMATCH:
+                p_comchannel->send("Argument type mismatch.\n");
+                break;
+            default:
+                p_comchannel->send("Non-specific error with the command.");
+        }
+    }
+    return b_command_recognised;
+}
+
+void CHumidityController::reset()
+{
+    for (int i = 0; i < CHANNEL_NUMBER; i++)
+    {
+        target_humidity = DISABLE_TARGET;
+        m_power_override[i] = DISABLE_OVERRIDE;
+        m_control_loop[i].reset();
+    }
+}
+
+void CHumidityController::sendStatus(IComChannel *p_comchannel)
+{
+    etl::string<MAX_STRING_SIZE> message;
+    char value[10];
+    // send target humidity
+    message.assign("Target:      ");
+    sprintf(value, "%2.1f", target_humidity);
+    message.append(value);
+    p_comchannel->send(message);
+    // Send actual humidity
+    message.assign("Humidity: ");
+    sprintf(value, "%2.1f, ", hum_sensor->getHumidity());
+    message.append(value);
+    p_comchannel->send(message);
+    // Send heater power
+    message.assign("Power:       ");
+    float power;
+    for (int i = 0; i < CHANNEL_NUMBER - 1; i++)
+    {
+        if (m_power_override[i] == DISABLE_OVERRIDE)
+        {
+            power = mp_hw->getHardPwmOutput(i);
+        }
+        else
+        {
+            power = m_power_override[i];
+        }
+        sprintf(value, "%4.1f, ", power);
+        message.append(value);
+    }
+    if (m_power_override[CHANNEL_NUMBER - 1] == DISABLE_OVERRIDE)
+    {
+        power = mp_hw->getHardPwmOutput(CHANNEL_NUMBER - 1);
+    }
+    else
+    {
+        power = m_power_override[CHANNEL_NUMBER - 1];
+    }
+    sprintf(value, "%4.1f\n", power);
+    message.append(value);
+    p_comchannel->send(message);
+}
+
+/**
+ * @brief Set humidity value
```

Comment:
Incomplete doxy header. Should have 
`@param p_command` field. 
We need to make sure you have right setup to do automatic doxygen headers generation.